src/devtools/store.js

```diff
@@ -420,6 +422,28 @@ export default class Store extends EventEmitter {
     this.emit('isProfiling');
   }
 
+  toggleIsCollapsed(id: number, isCollapsed: boolean): void {
+    const element = this.getElementByID(id);
+    if (element !== null) {
+      element.isCollapsed = isCollapsed;
+
+      const weightDelta = isCollapsed ? 1 - element.weight : element.weight - 1;
+
+      this._numElements += weightDelta;
```

Comment:
This logic might be a little easier to understand if you instead make a synthetic id-0 root node that has all the React roots as its children? Then you also won't need the special case in onBridgeOperations.

(This also seems good though. It wasn't obvious to me the purpose of this property though – renaming to `_weightAcrossRoots` or similar might be easier to understand.)

Reply:
Sure. I don't mind renaming it. There's an inline comment that explains what it is too.

> Total number of visible elements (within all roots).
> Used for windowing purposes.